### PR TITLE
Drop OpenSSL 0.9.8

### LINKED ISSUE #2963: 0.9.8 fallout ticket
This is a place to a) collect people's complaints on our dropping 0.9.8 and evaluate the merits thereof. b) if we evaluate the merits of their complaints and find them wanting, in 2 weeks we will delete the 0.9.8 code from cryptography. And it will be glorious

### PR DESCRIPTION
**Pull request by alex**
Fixes #2963

### CODE REVIEW

**Comment by Ayrx** [comment]
:cake:

File: docs/faq.rst [hunk 1]
```diff
@@ -40,19 +40,15 @@ If you have no other libraries using OpenSSL in your process, or they do not
...
+Installing cryptography with OpenSSL 0.9.8 fails
+------------------------------------------------
 
 The OpenSSL project has dropped support for the 0.9.8 release series. Since it
 is no longer receiving security patches from upstream, ``cryptography`` is also
 dropping support for it. To fix this issue you should upgrade to a newer
 version of OpenSSL (1.0.1 or later). This may require you to upgrade to a newer
 operating system.
 
-For the 1.4 release, you can set the ``CRYPTOGRAPHY_ALLOW_OPENSSL_098``
```
Comment (reaperhulk) [inline docs/faq.rst]: I don't want to remove the information about how to do this in 1.4 from our docs until we're ready to ship 1.5 (a small negative of using evergeen docs).

File: src/cryptography/hazmat/backends/openssl/backend.py [hunk 2]
```diff
@@ -1142,19 +1082,10 @@ def load_pem_public_key(self, data):
     def load_der_private_key(self, data, password):
         # OpenSSL has a function called d2i_AutoPrivateKey that can simplify
         # this. Unfortunately it doesn't properly support PKCS8 on OpenSSL
-        # 0.9.8 so we can't use it. Instead we sequentially try to load it 3
+        # 0.9.8 so we can't use it. Instead we sequentially try to load it 2
```
Comment (reaperhulk) [inline src/cryptography/hazmat/backends/openssl/backend.py]: We might be able to switch to d2i_AutoPrivateKey here now, but that should be a separate PR.
